pkg/resource/deploy/source_eval.go

```diff
@@ -1159,6 +1162,16 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 		outputs = filtered
 	}
 
+	// TODO(@platform): Currently component resources ignore the "ignoreChanges" option.
+	// Revisit these semantics in Pulumi v4.0
+	// See this issue for more: https://github.com/pulumi/pulumi/issues/9704
+	if !custom && len(ignoreChanges) > 0 {
+		rm.ctx.Diag.Warningf(diag.Message(
+			result.State.URN,
+			"The option 'ignoreChanges' has no effect on components. This may be changed in a future version.",
```

Comment:
I'd drop the comment about changing in future versions.

Reply:
Thought that might be divisive :)
Sounds good!